boruta/script.whizzml

```diff
@@ -0,0 +1,391 @@
+        obj-f (filter-map (lambda (x _) (= (get-in fields
+                                                   [x "name"])
+                                           obj-name))
+                          f-id-names)
+        obj-id (head (keys obj-f))
+        f-id-names (filter-map (lambda (x _) (!= x obj-id)) f-id-names)
+        f-ids (keys f-id-names)
+        input-fields (cons obj-id f-ids)
+        f-ids (repeat-fields f-ids 5)
+        f-names (map (lambda (x) (str "shadow " (get f-id-names x)))
+                     f-ids)
+        flatline-fs (map (lambda (x) [(flatline
+                                        "(weighted-random-value {{x}})")
+                                      (str "shadow " (get f-id-names x))])
+                         f-ids)
+        new-fields (lambda (x) (assoc {} "field" (head x)
+                                         "name" (head (tail x))))
+        sh-ds-id (create-and-wait-dataset {"origin_dataset"
+                                           ds-id
+                                           "input_fields"
+                                           input-fields
+                                           "new_fields"
+                                           (map new-fields flatline-fs)})
+        sh-ds (fetch sh-ds-id)
+        sh-fields (get sh-ds "fields")
+        sh-f-ids (filter (lambda (x) (member? (get-in sh-fields [x "name"])
+                                              f-names))
+                         (keys sh-fields))
+        old-fields (filter-map (lambda (x _) (not (member? (get-in sh-fields
+                                                           [x "name"])
+                                                  f-names)))
+                               sh-fields))
+    [sh-ds-id sh-f-ids old-fields]))
+
+;; repeat-fields
+;; Repeats the field ID until the number of elements is the one chosen
+;;
+;; Inputs:
+;;   f-ids: (list) List of field IDs
+;;   min-count: (integer) Minimum number of field IDs
+;;
+;; Output: (list) List of field IDs
+(define (repeat-fields f-ids min-count)
+  (if (< (count f-ids) min-count)
+      (repeat-fields (append f-ids (nth f-ids (rem min-count (count f-ids))))
+                     min-count)
+      f-ids))
+
+;; random-forest
+;; Create a random forest from the dataset
+;;
+;; Inputs:
+;;   ds-id: (dataset-id) Dataset ID
+;;   obj: (string) Objective field ID
+;; Output: (ensemble-id) Random Forest ID
+(define (random-forest ds-id obj)
+  (create-and-wait-ensemble {"dataset" ds-id
+                             "objective_field" obj
+                             "randomize" true
+                             "sample_rate" 0.8
+                             "seed" "BigML Boruta"}))
+
+;; assoc+
+;;
+;; Adds the value in the key-value pair to the one existing in the accumulator
+;;
+;; Input:
+;;   acc: (map) accumulator map
+;;   key: (string) key to accumulate
+;;   value: (number) value to be added
+(define (assoc+ acc key value)
```

Comment:
I believe this is just `(assoc acc key (+ value (get acc key 0)))`  Correct me if I'm wrong, @jaor .